lib/kramdown-rfc2629.rb

```diff
@@ -102,10 +102,13 @@ def parse_xref
             when /\A(.*) \((#{SECTIONS_RE})\)\z/
               href = $1
               handle_bares($2, attr, "parens", href)
-            when /\A([\w.]+)<(.*)\z/
+            when /\A(.+)<(.+)\z/
               href = $2
               attr['section'] = $1
               attr['sectionFormat'] = 'bare'
+            when /\A(.+)<\z/
+              href= $1
+              attr['format'] = 'counter'
```

Comment:
While we are at it, should we invent syntax for format="title"?

Reply:
`{{...>Title}}`.  Seems like the combinations are hard to work through though...

* `{{foo<>Title}}` seems unnecessary as a title on a bare counter is odd.

* `{{1.2<RFC2119>Title}}` is plausible.

* `{{!NORM=RFC2119>Title}}` seems OK.

* `{{Section 3 of RFC8446>Title}}` maybe as well.

WDYT?  Or do you think that a generic syntax would be better?

Reply:
Sorry, I should have read up on what format="title" does.  Now that I have and I'm over the initial repulsion, it seems like it matters what sectionFormat does here.  If it does nothing (which seems likely given the constraints), then maybe this can be as simple as:

* `{{<foo}}` to get a local section number rather than `{{foo}}`, because I can't quite reconcile this with the underlying grammar otherwise.
* `{{<<foo}}` to get a local section title, or a title for a \<reference>.
* `{{1.2<reference}}` to get a bare section number to a \<reference>.
* `{{1.2<<x}}` makes no sense and would be illegal.

xml2rfc will complain if "foo" isn't local for `{{<foo}}`, but it's OK to pass down invalid stuff.

Does that sound better?

Reply:
The original sin was to mix up internal references with external ones.  Staying within that space, I think your proposal is pretty good.  Keeps the `!?-` in one place for and update to `xml_from_sections` in `bin/kramdown-rfc2629` (ouch).

{{G<<RFC8610}} might become possible in the future (if that info ever migrates into RFC8610.json or a similar resource), so it is good to already have syntax in mind for that.